pyodide-build/pyodide_build/xbuildenv.py

```diff
@@ -0,0 +1,332 @@
+        """
+        versions = []
+        for version_dir in self.env_dir.glob("*"):
+            if not version_dir.is_dir() or str(version_dir) == str(self.symlink_dir):
+                continue
+
+            versions.append(version_dir.name)
+
+        return sorted(versions)
+
+    def use_version(self, version: str) -> None:
+        """
+        Select the xbuildenv version to use.
+
+        This creates a symlink to the selected version in the xbuildenv directory.
+
+        Parameters
+        ----------
+        version
+            The version of xbuildenv to use.
+        """
+        logger.info(f"Using Pyodide cross-build environment version: {version}")
+
+        version_path = self._path_for_version(version)
+        if not version_path.exists():
+            raise ValueError(
+                f"Cannot find cross-build environment version {version}, available versions: {self.list_versions()}"
+            )
+
+        symlink_dir = self.symlink_dir
+
+        if symlink_dir.exists():
+            if symlink_dir.is_symlink():
+                # symlink to a directory, expected case
+                symlink_dir.unlink()
+            elif symlink_dir.is_dir():
+                # real directory, for backwards compatibility
+                shutil.rmtree(symlink_dir)
+            else:
+                # file. This should not happen unless the user manually created a file
+                # but we will remove it anyway
+                symlink_dir.unlink()
+
+        symlink_dir.symlink_to(version_path)
+
+    def install(
+        self,
+        version: str | None = None,
+        *,
+        url: str | None = None,
+        skip_install_cross_build_packages: bool = False,
+    ) -> Path:
+        """
+        Install cross-build environment.
+
+        Parameters
+        ----------
+        version
+            The version of the cross-build environment to install. If not specified,
+            use the same version as the current version of pyodide-build.
+            # TODO: installing the different version is not supported yet
+        url
+            URL to download the cross-build environment from.
+            The URL should point to a tarball containing the cross-build environment.
+            This is useful for testing unreleased version of the cross-build environment.
+
+            Warning: if you are downloading from a version that is not the same
+            as the current version of pyodide-build, make sure that the cross-build
+            environment is compatible with the current version of Pyodide.
+        skip_install_cross_build_packages
+            If True, skip installing the cross-build packages. This is mostly for testing purposes.
+
+        Returns
+        -------
+        Path to the root directory for the cross-build environment.
+        """
+
+        if url:
+            if version:
+                raise ValueError("Cannot specify both version and url")
```

Comment:
Nit: I like to separate the error case into its own block:
```suggestion
        if url and version:
            raise ValueError("Cannot specify both version and url")
        if url:
```